### ISSUE TITLE
Testing typing with mypy

### ISSUE THREAD

**Issue by ikonst** [python/mypy#6115]
Occasionally one would like to test a third party library's typing/stubs using mypy. While testing typing can seem a foolish endeavor ("testing the tests"), in some cases it can be justified:
1. when the typing is complex (generics etc.) and you want to ensure common user patterns are not flagged as errors
2. when developing a mypy plugin for the third party library

Personally I've encountered the need when working on pynamodb typestubs and mypy plugin.

mypy has enviable tooling for data-driven tests, but leveraging it in a third party library requires some hacks. Here's a recipe I worked out:
- tests/conftest.py
  ```py
  pytest_plugins = [
     'mypy.test.data',
  ]

  import os.path
  os.environ['MYPY_TEST_PREFIX'] = os.path.dirname(__file__)  # HACK
  ```

- tests/test_mypy.py
  ```py
  from mypy.test.testcheck import TypeCheckSuite
  TypeCheckSuite.files = [  # HACK
      'check-thirdparty.test',
  ]
  ```

- tests/test-data/unit/check-thirdparty.test
  ```
  [case foo]
  class Foo:
    bar = 'baz'
  
  reveal_type(Foo.bar)  # E: Revealed type is 'builtins.str'
  ```

Can we can make it friendlier to third-party consumption?

**Comment by ikonst** [comment]
BTW, here's what I came up while working on pynamodb-mypy:
https://github.com/pynamodb/PynamoDB/pull/576

As you can see, it's very basic. I wonder what stuff I'm missing from mypy's suite that would be useful for third-party testing.

**Comment by JukkaL** [comment]
We probably won't have time to work on this, and there hasn't been recent activity, so I'm closing this.
